src/ui/src/builder/useComponentActions.ts

```diff
@@ -688,6 +696,73 @@ export function useComponentActions(wf: Core, ssbm: BuilderManager) {
 		wf.sendComponentUpdate();
 	}
 
+	/**
+	 * Add an out
+	 */
+	function addOut(
+		componentId: Component["id"],
+		out: Component["outs"][number],
+	) {
+		const component = wf.getComponentById(componentId);
+		if (!component) return;
+		const transactionId = `add-${componentId}-out-${out.outId}-${out.toNodeId}`;
+		ssbm.openMutationTransaction(transactionId, `Add out`, true);
+		ssbm.registerPreMutation(component);
+
+		component.outs = [...(component.outs ?? []), out];
+
+		ssbm.registerPostMutation(component);
+		ssbm.closeMutationTransaction(transactionId);
+		wf.sendComponentUpdate();
+	}
+
+	/**
+	 * Remove an out
+	 */
+	function removeOut(
+		componentId: Component["id"],
+		out: Component["outs"][number],
+	) {
+		const component = wf.getComponentById(componentId);
+		if (!component) return;
+		const transactionId = `edit-${componentId}-out-${out.outId}-${out.toNodeId}`;
+		ssbm.openMutationTransaction(transactionId, `Edit out`, true);
+		ssbm.registerPreMutation(component);
+
+		component.outs = [
+			...component.outs.filter(
+				(o) => !(out.outId === o.outId && out.toNodeId === o.toNodeId),
+			),
+		];
```

Comment:
You don't have to "clone" the array again
```suggestion
		component.outs = component.outs.filter(
			(o) => !(out.outId === o.outId && out.toNodeId === o.toNodeId),
		);
```

Reply:
Yes, completely unnecessary, well spotted.